src/libtriton/arch/irBuilder.cpp

```diff
@@ -292,12 +242,57 @@ namespace triton {
 
 
     void IrBuilder::removeSymbolicExpressions(triton::arch::Instruction& inst, std::set<triton::ast::AbstractNode*>& uniqueNodes) {
-      for (auto it = inst.symbolicExpressions.begin(); it != inst.symbolicExpressions.end(); it++) {
+      for (auto it = inst.symbolicExpressions.cbegin(); it != inst.symbolicExpressions.cend(); it++) {
         this->astGarbageCollector.extractUniqueAstNodes(uniqueNodes, (*it)->getAst());
         this->symbolicEngine->removeSymbolicExpression((*it)->getId());
       }
       inst.symbolicExpressions.clear();
     }
 
+
+    template <typename T>
+    void IrBuilder::collectNodes(std::set<triton::ast::AbstractNode*>& uniqueNodes, T& items) const {
+      for (auto it = items.cbegin(); it != items.cend(); it++)
+        this->astGarbageCollector.extractUniqueAstNodes(uniqueNodes, std::get<1>(*it));
+      items.clear();
+    }
+
+
+    void IrBuilder::collectNodes(std::set<triton::ast::AbstractNode*>& uniqueNodes, std::vector<triton::arch::OperandWrapper>& operands, bool gc) const {
+      for (auto it = operands.begin(); it != operands.end(); it++) {
+        if (it->getType() == triton::arch::OP_MEM) {
+          if (gc)
+            this->astGarbageCollector.extractUniqueAstNodes(uniqueNodes, it->getMemory().getLeaAst());
+          it->getMemory().setLeaAst(nullptr);
+        }
+      }
+    }
+
+
+    template <typename T>
+    void IrBuilder::collectUnsymbolizedNodes(T& items) const {
+      T newItems;
+
+      for (auto it = items.cbegin(); it != items.cend(); it++) {
+        if (std::get<1>(*it) && std::get<1>(*it)->isSymbolized() == true)
+          newItems.insert(*it);
+      }
+
+      items.clear();
+      items = newItems;
```

Comment:
Here you could "move" the new container and it would be done in O(1) instead of O(n)